offlineimap/repository/Base.py

```diff
@@ -182,12 +182,18 @@ def sync_folder_structure(self, dst_repo, status_repo):
         # Create hashes with the names, but convert the source folders
         # to the dest folder's sep.
         src_hash = {}
+        src_list = []
         for folder in src_folders:
-            src_hash[folder.getvisiblename().replace(
+            foldername = folder.getvisiblename()
+            src_list.append(foldername)
+            src_hash[foldername.replace(
                 src_repo.getsep(), dst_repo.getsep())] = folder
         dst_hash = {}
+        dst_list = []
         for folder in dst_folders:
-            dst_hash[folder.getvisiblename().replace(
+            foldername = folder.getvisiblename()
+            dst_list.append(foldername)
+            dst_hash[foldername.replace(
                 dst_repo.getsep(), src_repo.getsep())] = folder
```

Comment:
So, a name in
- src_list might diverge from src_hash
- dst_list might diverge from src_hash
for this name....